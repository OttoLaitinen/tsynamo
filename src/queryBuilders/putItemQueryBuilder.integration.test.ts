import { DDB, TEST_DATA } from "../../test/testFixture";
import { getDDBClientFor, startDDBTestContainer } from "../../test/testUtil";
import { Tsynamo } from "./../index";

describe("GetItemQueryBuilder", () => {
  let tsynamoClient: Tsynamo<DDB>;

  const itemToPut = {
    userId: "333",
    dataTimestamp: 222,
    someBoolean: true,
  };

  beforeAll(async () => {
    const testContainer = await startDDBTestContainer();

    tsynamoClient = new Tsynamo<DDB>({
      ddbClient: await getDDBClientFor(testContainer),
    });
  });

  it("handles a simple put query", async () => {
    let result = await tsynamoClient
      .getItemFrom("myTable")
      .keys({
        userId: itemToPut.userId,
        dataTimestamp: itemToPut.dataTimestamp,
      })
      .execute();

    expect(result).toBeUndefined();

    await tsynamoClient.putItem("myTable").item(itemToPut).execute();

    result = await tsynamoClient
      .getItemFrom("myTable")
      .keys({
        userId: "333",
        dataTimestamp: 222,
      })
      .execute();

    expect(result).toBeDefined();
    expect(result).toEqual(itemToPut);
  });

  it("handles ReturnValues option", async () => {
    let result = await tsynamoClient
      .putItem("myTable")
      .item(itemToPut)
      .returnValues("ALL_OLD")
      .execute();

    expect(result).toBeUndefined();

    result = await tsynamoClient
      .putItem("myTable")
      .item({
        ...itemToPut,
        tags: ["kissa"],
      })
      .returnValues("ALL_OLD")
      .execute();

    expect(result).toEqual(itemToPut);
  });
});
