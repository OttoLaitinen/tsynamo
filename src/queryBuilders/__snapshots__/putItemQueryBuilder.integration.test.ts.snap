// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PutItemQueryBuilder > handles 'contains' ConditionExpression 1`] = `
{
  "dataTimestamp": 212,
  "tags": [
    "cats",
  ],
  "userId": "333",
}
`;
